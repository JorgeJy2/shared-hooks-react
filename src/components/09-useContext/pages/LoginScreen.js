import React, { useContext } from 'react'

import '../../01-useState/counter.css';
import { UserContext } from '../context/UserContext';

export const LoginScreen = () => {

    const { setUser } = useContext(UserContext);
  
    const handlerLogin = ()=>{
        setUser({
            name: 'jorge',
            email: 'jorge@gmail.com'
        });
    }
    return (
        <div className="container">
            <h1>Login</h1>
            <hr />
            <button onClick={handlerLogin} className="btn btn-primary" >Login</button>
        </div>
    );
}
