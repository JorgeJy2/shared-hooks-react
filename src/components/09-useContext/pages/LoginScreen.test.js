import { mount } from "enzyme";
import { UserContext } from "../context/UserContext";
import { LoginScreen } from "./LoginScreen";

describe('test login', () => {
    const user = {
        name: 'jorge',
        email: 'jorge@gmail.com'
    }

    const setUser = jest.fn(() => { });

    const wrapper = mount(
        <UserContext.Provider
            value={
                {
                    user,
                    setUser
                }}>
            <LoginScreen />
        </UserContext.Provider>
    );


    test('should snaptshot ', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('should login ', () => {

        wrapper.find('button').simulate('click');
        expect(setUser).toBeCalled();
        expect(setUser).toBeCalledWith({
            name: 'jorge',
            email: 'jorge@gmail.com'
        });


    });



});