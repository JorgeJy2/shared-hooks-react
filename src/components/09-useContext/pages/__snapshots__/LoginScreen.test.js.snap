// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login should snaptshot  1`] = `
<div
  className="container"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
